tools/execsnoop.py

```diff
@@ -105,7 +106,17 @@
 {
     // create data here and pass to submit_arg to save stack space (#555)
     struct data_t data = {};
+    struct task_struct *task;
+    struct task_struct *real_parent_task;
+    u32 ppid;
+
     data.pid = bpf_get_current_pid_tgid() >> 32;
+
+    task = (struct task_struct *)bpf_get_current_task();
+    bpf_probe_read(&real_parent_task, sizeof(real_parent_task), &task->real_parent);
+    bpf_probe_read(&ppid, sizeof(ppid), &real_parent_task->tgid);
+    data.ppid = ppid;
```

Comment:
You shouldn't need explicit calls to `bpf_probe_read` here:
```c
daat.ppid = task->real_parent->tgid;
```
bcc will handle the rewrite.

Reply:
ok, good to know, I tried that too and got similar results.

Reply:
Yep, that wouldn't change the issue with the parent PID. It translates to the same bytecode.